crates/bevy_ecs/src/query/iter.rs

```diff
@@ -121,11 +122,11 @@ impl<'w, 's, D: QueryData, F: QueryFilter> QueryIter<'w, 's, D, F> {
         let entities = table.entities();
         for row in rows {
             // SAFETY: Caller assures `row` in range of the current archetype.
-            let entity = entities.get_unchecked(row);
+            let entity = unsafe { entities.get_unchecked(row) };
             let row = TableRow::from_usize(row);
             // SAFETY: set_table was called prior.
             // Caller assures `row` in range of the current archetype.
-            if !F::filter_fetch(&mut self.cursor.filter, *entity, row) {
+            if unsafe { !F::filter_fetch(&mut self.cursor.filter, *entity, row) } {
```

Comment:
Same with the match, having an unsafe block as the conditional in an if expression is not great for readability.